enterprise/reporting/error-index/worker.go

```diff
@@ -0,0 +1,386 @@
+package error_index
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"io"
+	"os"
+	"path"
+	"path/filepath"
+	"sort"
+	"strconv"
+	"strings"
+	"sync"
+	"time"
+
+	"github.com/samber/lo"
+	"github.com/xitongsys/parquet-go/parquet"
+	"github.com/xitongsys/parquet-go/writer"
+
+	"github.com/rudderlabs/rudder-go-kit/bytesize"
+	"github.com/rudderlabs/rudder-go-kit/config"
+	"github.com/rudderlabs/rudder-go-kit/logger"
+	"github.com/rudderlabs/rudder-go-kit/stats"
+	kitsync "github.com/rudderlabs/rudder-go-kit/sync"
+	"github.com/rudderlabs/rudder-server/jobsdb"
+	"github.com/rudderlabs/rudder-server/utils/misc"
+)
+
+type jobPayload struct {
+	job     *jobsdb.JobT
+	payload payload
+}
+
+type worker struct {
+	sourceID    string
+	workspaceID string
+
+	log          logger.Logger
+	statsFactory stats.Stats
+
+	jobsDB        jobsdb.JobsDB
+	configFetcher configFetcher
+	uploader      uploader
+
+	lifecycle struct {
+		ctx    context.Context
+		cancel context.CancelFunc
+	}
+
+	limiterGroup sync.WaitGroup
+	limiter      struct {
+		fetch  kitsync.Limiter
+		upload kitsync.Limiter
+		update kitsync.Limiter
+	}
+
+	now            func() time.Time
+	lastUploadTime time.Time
+
+	config struct {
+		parquetParallelWriters, parquetRowGroupSizeInMB, parquetPageSizeInKB misc.ValueLoader[int64]
```

Comment:
All sizes should be assumed to be in bytes, so no `InMB`/`InKB` prefixes should exist